### ISSUE TITLE
[HOLD for payment 2025-01-18] [$250] Search - Workspace chat preview, flickers between last message and information message

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53361]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.69-1
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Y
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Open the staging.new.expensify.com website.
2. Tap on "Settings" and select "Workspaces"
3. Open any workspace and tap on "Members"
4. Tap on "Invite Member"
5. Invite any member to workspace.
6. Return to LHN.
7. Open the workspace individual chat created for this invited member.
8. Send any message on it.
9. Tap on the search icon.
10. Start writing a search related to this last chat.
11. Verify that with every letter added on the search input, this chat preview keeps displaying the last sent message.

## Expected Result:
The preview of the individual chat created for an invited workspace member, should keep displaying the last sent message when writing on the search input.

## Actual Result:
Chat preview of individual chat created for an invited workspace member, flickers between the last sent message and the workspace information message with every letter added on the search input.

## Workaround:
Unknown

## Platforms:

- [x] Android: Standalone
- [x] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/b99fa083-f588-49e1-91ab-138716a56f76
  
</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021863580183549143544</li>
        <li>Upwork Job ID: 1863580183549143544</li>
        <li>Last Price Increase: 2024-12-16</li>
<li>Automatic offers: </li>
<ul>
<li>rojiphil | Reviewer | 105395423</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @isabelastisser</details>

**Comment by mountiny** [comment]
This is a minor ux issue, feels like the state is changing between optimistic offline and the response from the server

Demoted

**Comment by mountiny** [comment]
cc @luacmartins @lakchote you might know the best about changes related to this area of code

**Comment by Beamanator** [comment]
Demoting makes sense to me 👍  making this external since the flicker sounds like a UI-only bug

**Comment by luacmartins** [comment]
I agree that we should remove the mutations.

>I'm pretty sure we can avoid performance issues without introducing mutations. How many options are we processing here?

This function is used by the LHN as well, so we are processing the full LHN results for users. Not sure how large that data set is though.

**Comment by UmairMukhtar** [comment]
https://github.com/user-attachments/assets/6659b40d-f930-4c60-bd45-0de608e5f763

This issue does not exist anymore and may be closed

**Comment by UmairMukhtar** [comment]
Contributor details
Your Expensify account email: umairmukhtar27+@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~0102d6a693d924b83b

**Comment by isabelastisser** [comment]
@rojiphil, any updates here?

**Comment by rojiphil** [comment]
@isabelastisser Sorry for the delay here. I have picked this for review now and will update sometime today.

**Comment by rojiphil** [comment]
@bernhardoj I am not able to reproduce this issue with the latest main. Can you reproduce? 

However, I do find the mutation problem interesting. If mutations are causing problems, they must be manifested elsewhere, too. Did you find any other issues due to these mutations? 

https://github.com/user-attachments/assets/45fee654-dbe1-4e34-9822-c97e4797e621

**Comment by bernhardoj** [comment]
Yeah, I can't see the flickering anymore too, but you can see that it still shows the last message of the policy expense chat even though we set `forcePolicyNamePreview` as true.
https://github.com/Expensify/App/blob/7925c9b555af60156c62947ae63b517d6c31587f/src/libs/OptionsListUtils.ts#L1297-L1306

If we don't mutate the object, then we will see the policy name as the subtitle.

This is another issue caused by object mutation: https://github.com/Expensify/App/issues/52975

**Comment by isabelastisser** [comment]
@rojiphil, please provide an update. Thanks!

**Comment by rojiphil** [comment]
> Yeah, I can't see the flickering anymore too, but you can see that it still shows the last message of the policy expense chat even though we set `forcePolicyNamePreview` as true.

@bernhardoj The expected result here is to display the last message text which is what it does. So, I am not sure what the issue is here. Can you please explain a bit more on this?

> This is another issue caused by object mutation: #52975

That is another issue and you are already handling it there. But, I am curious to know if we can reproduce a problem in our context here with any other variable e.g. `isSelected`, `isBold` as referred in your proposal.

**Comment by bernhardoj** [comment]
> The expected result here is to display the last message text which is what it does. So, I am not sure what the issue is here. Can you please explain a bit more on this?

Based on the code, the expected result is to show the policy name as explained in my previous comment. You can revert https://github.com/Expensify/App/pull/53293 and see it will show the policy name.

>  But, I am curious to know if we can reproduce a problem in our context here with any other variable e.g. isSelected, isBold as referred in your proposal.

You can change this to true
https://github.com/Expensify/App/blob/3982ed03ff644e7a723e8d4131535a2cceb9488d/src/components/Search/SearchRouter/SearchRouterList.tsx#L159

and search for a DM and the title will be bold even though not bold on the LHN.

<img width="1238" alt="image" src="https://github.com/user-attachments/assets/b6fa2218-ff71-464e-842b-fedf768313c0">

**Comment by isabelastisser** [comment]
@rojiphil, please share an update. Thanks!

**Comment by rojiphil** [comment]
> Based on the code, the expected result is to show the policy name as explained in my previous comment. You can revert #53293 and see it will show the policy name.

@bernhardoj The mentioned PR has been in production for the last 15 days and seems acceptable to all. So, I am still not sure why a revert is needed. Can you please point out the relevant PR/issue where the expected result is to show the policy name?

**Comment by bernhardoj** [comment]
> So, I am still not sure why a revert is needed.

I'm not saying that the solution here is to revert, but fix the mutation issue. I just mentioned that you can revert and see the behavior before that PR to prove the issue that is caused by mutating an object.

> Can you please point out the relevant PR/issue where the expected result is to show the policy name?

I'm not aware of this, but the code was added in https://github.com/Expensify/App/pull/3540.

If the current behavior is acceptable, then we can close this and wait until another unexpected issue caused by the mutation side-effect comes up.

**Comment by lakchote** [comment]
> If the current behavior is acceptable, then we can close this and wait until another unexpected issue caused by the mutation side-effect comes up.
>

This is not an acceptable behavior to have a flickering UI (bug starts happening at `0:43` in the video for those finding the issue).

We need to address this.

**Comment by bernhardoj** [comment]
Oh, I don't mean the flickering behavior. I and @rojiphil can't see the flickering anymore, but the mutation still causes a different expected behavior shown from the code which I explained [here](https://github.com/Expensify/App/issues/53361#issuecomment-2526977030).

**Comment by rojiphil** [comment]
> I just mentioned that you can revert and see the behavior before that PR to prove the issue that is caused by mutating an object.

Ok. I will check this today and update.

**Comment by Beamanator** [comment]
Love where we got with this issue 👍 

Though I def believe we need to update the issue title & OP reproduction steps...

So it sounds like the new problem is here: https://github.com/Expensify/App/issues/53361#issuecomment-2526977030

Can y'all confirm & let me know what reproduction steps y'all are following so I can update the OP?

**Comment by bernhardoj** [comment]
Here are the steps:
Prerequisite:
- have a workspace
- have at least 1 other member
- send a message at least once to the member workspace chat
1. Open the search modal
2. Search for the member workspace chat
3. Observe that the subtitle shows the last message instead of the workspace name

NOTE: For user's own workspace chat, the workspace name is shown, so this issue only happens with member workspace chat

**Comment by bernhardoj** [comment]
PR is ready

cc: @rojiphil

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by isabelastisser** [comment]
Payment summary:

@rojiphil $250, payment being handled in Upwork. [automatic offer](https://www.upwork.com/nx/wm/offer/105395423) C+
@bernhardoj requires payment through NewDot Manual Requests $250 for proposal, payment pending in ND.

**Comment by bernhardoj** [comment]
Requested in ND.

**Comment by rojiphil** [comment]
@isabelastisser Accepted the offer. Just the payment is pending. Thanks.

**Comment by JmillsExpensify** [comment]
$250 approved for @bernhardoj
